contracts/ssnlist.scilla

```diff
@@ -767,19 +773,29 @@ procedure CalcStakeDelegPerCycle(tmp_arg: TmpArg)
     buf_opt <- buff_deposit_deleg[deleg][ssn_operator][last2_reward_cycle];
     comb_opt = option_add cur_opt buf_opt;
 
+    last_amt_o <- deleg_stake_per_cycle[deleg][ssn_operator][last_reward_cycle];
+    last_amt = option_uint128_value last_amt_o;
+    staking_and_rewards_per_cycle_for_ssn_opt <- stake_ssn_per_cycle[ssn_operator][reward_cycle];
+    (* the rewards for this cycle also computed within this procedure, so we no long keep histric data *)
+    delete deleg_stake_per_cycle[deleg][ssn_operator][last_reward_cycle];
```

Comment:
This line is redundant.

Reply:
Hmm no, because I am writing the data for this cycle, while I am deleting last cycle.